### PR TITLE
Improves issue#2

### LINKED ISSUE #2: Save Image Link
#### As a user I want to save a link to an image.

Acceptance criteria:
- [x] The link is entered through a form.
- [x] After the form is saved, the link is persisted in the database.
- [x] After the form is submitted, the user is redirected to a page displaying the image.

Dependencies:
- Deployment Environment

__Note__: Many image sharing sites will permit one to upload images, or
download images when provided a link. For this project we will only work with
the link, i.e., URL, to various images.

### PR DESCRIPTION
**Pull request by wpliao1989**
Closes #2 

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [x] I have replaced `X` above with the associated issue number.
- [x] I have changed the issue's label to "pull request".
- [x] I have assigned the designated reviewer to this PR.
- [x] I have set the appropriate label on this PR.

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

**Comment by wpliao1989** [comment]
Added 1 test case and refactored a few things.

File: app/controllers/images_controller.rb [hunk 1]
```diff
@@ -4,8 +4,8 @@ def new
   end
 
   def create
-    image = Image.create(image_params)
-    redirect_to action: :show, id: image.id
+    @image = Image.create!(image_params)
+    redirect_to @image
```
Comment (bboe) [inline app/controllers/images_controller.rb]: Can we add a `notice` type flash message to indicate that the image was successfully created? Providing little messages like this do a lot to provide a better user experience.

File: test/controllers/images_controller_test.rb [hunk 2]
```diff
@@ -9,17 +9,21 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
     url = 'http://www.appfolio.com/images/html/apm-logo-v2.png'
-    assert_not Image.exists?(url: url)
-    post images_url image: { url: url }
-    assert_response :found
-    assert Image.exists?(url: url)
+    assert_difference('Image.count') do
+      post images_url image: { url: url }
+    end
+    assert_redirected_to image_url(Image.last)
   end
```
Comment (bboe) [inline test/controllers/images_controller_test.rb]: After adding a flash, an additional side effect test will need to be here.

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -9,17 +9,21 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
 
   test 'should get show' do
-    url = 'http://www.appfolio.com/images/html/apm-logo-v2.png'
+    url   = 'http://www.appfolio.com/images/html/apm-logo-v2.png'
```
Comment (bboe) [inline test/controllers/images_controller_test.rb]: 👾

File: test/controllers/images_controller_test.rb [hunk 4]
```diff
@@ -9,17 +9,21 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
     image = Image.create(url: url)
     get image_url id: image.id
     assert_response :success
     assert_select 'p'
   end
+
+  test 'show not get show' do
+    assert_raises(ActiveRecord::RecordNotFound) { get image_url id: -1 }
```
Comment (bboe) [inline test/controllers/images_controller_test.rb]: Great test!

**Comment by bboe** [comment]
A few minor things to address.

File: test/controllers/images_controller_test.rb (hunk 3 above)
Comment (wpliao1989) [inline test/controllers/images_controller_test.rb]: Actually I did run rubocop and it didn't complain about this. I used RubyMine's auto formatting so it aligned this to the variable below. Should I changed it back?

Comment (bboe) [inline test/controllers/images_controller_test.rb]: Interesting. Well if rubocop didn't complain it should be fine.

**Comment by wpliao1989** [comment]
Updated

**Comment by bboe** [comment]
The changes look great.

How do you feel about the commit message? Pretend that I know nothing about this project, does that commit message tell me what is contained within it?

Please amend that commit message with something more descriptive and force push.

**Comment by wpliao1989** [comment]
Got it & updated

**Pull request merged**